lib/cluster.py

```diff
@@ -120,6 +122,88 @@ def reset(self, config):
         )
         assert(status == 0)
 
+        # Validate CBGT
+        if not self.validate_cbgt_pindex_distribution_retry():
+            self.save_cbgt_diagnostics()
+            raise Exception("Failed to validate CBGT Pindex distribution")
+
+        print(">>> Detected valid CBGT Pindex distribution")
+
+    def save_cbgt_diagnostics(self):
+        
+        # CBGT REST Admin API endpoint 
+        adminApi = Admin(self.sync_gateways[0])
+        cbgt_diagnostics = adminApi.get_cbgt_diagnostics()
+        cbgt_cfg = adminApi.get_cbgt_cfg()
+        
+        # dump raw diagnostics
+        # (unfortunately this doesn't parse in as json)
+        print("CBGT diagnostic output: {}".format(cbgt_diagnostics))
+
+        # dump cfg json 
+        pretty_print_json = json.dumps(cbgt_cfg, sort_keys=True, indent=4, separators=(',', ': '))
+        print("CBGT CFG output: {}".format(pretty_print_json))        
```

Comment:
Perhaps we should dump to a file in case this happens during a run with lots of output?